src/components/CytoscapeGraph/CytoscapeGraph.tsx

```diff
@@ -280,9 +285,10 @@ export class CytoscapeGraph extends React.Component<CytoscapeGraphProps, Cytosca
     // update the entire set of nodes and edges to keep the graph up-to-date
     cy.json({ elements: this.props.elements });
 
-    // update the layout if needed
+    // update the layout if needed and reset to default values
     if (this.updateLayout) {
       cy.layout(LayoutDictionary.getLayout(this.props.graphLayout)).run();
+      clearSelection();
```

Comment:
We don't want to reset the selection just because we changed the layout. In fact, its a nice feature that I could select a node, and change the layout from breadthfirst to Dagre and still see the same node selected as before. There is no reason why we would need or want the selection to go away if we simply change the layout.

We only want to clear the selection when we change the NAMESPACE (which is what is happening when we double-click a pentagon node) because in that case the node that is selected is probably going away anyway.